components/dialpad/sources/new-call-event/new-call-event.ts

```diff
@@ -0,0 +1,29 @@
+import { defineSource } from "@pipedream/types";
+import constants from "../../common/constants";
+import common from "../../common/common";
+
+export default defineSource({
+  ...common,
+  type: "source",
+  name: "New Call Event (Instant)",
+  key: "dialpad-new-call-event",
+  description: `Emit new call event subscription. 
+  See [Event doc](https://developers.dialpad.com/reference/webhook_call_event_subscriptioncreate)
+  and [webhook doc](https://developers.dialpad.com/reference/webhookscreate)`,
+  version: "0.0.1",
+  dedupe: "unique",
+  methods: {
+    ...common.methods,
+    getPath() {
+      return constants.EVENT_TYPE.NEW_CALL.path;
+    },
```

Comment:
When I tried to deploy this source, I received an error:
```
    "error": {
      "code": 400,
      "errors": [
        {
          "domain": "global",
          "message": "Call states must be provided.",
          "reason": "badRequest"
        }
      ],
      "message": "Call states must be provided."
    }
```
Dialpad's [Call Event -- Create](https://developers.dialpad.com/reference/webhook_call_event_subscriptioncreate) endpoint seems to require that the `call_states` parameter is defined and not empty. For example:
```js
    getPayload() {
      return {
        call_states: [
          "all",
        ],
      };
    },
```
Should this source emit events for only *new* calls or for *all* [call states](https://developers.dialpad.com/docs/call-events-logging#call-states)? If an event should be emitted only when a new call is created, I think we could use either the `calling` or `hangup` call state instead of `all`.
```js
      call_states: [ "calling" ],
```

Reply:
This is done. Unfortunately, I am unable to test.